src/test/kotlin/coverosR3z/authentication/AuthenticationPersistenceTests.kt

```diff
@@ -87,15 +88,17 @@ class AuthenticationPersistenceTests {
      */
     @Test
     fun testCorruptingSessionDataWithMultiThreading() {
-        val listOfThreads = mutableListOf<Thread>()
+        val ap = AuthenticationPersistence(PureMemoryDatabase())
+        val listOfThreads = mutableListOf<Future<*>>()
+        val cachedThreadPool: ExecutorService = Executors.newCachedThreadPool(Executors.defaultThreadFactory())
         val numberNewSessionsAdded = 20
         for(i in 1..numberNewSessionsAdded) { // each thread calls the add a single time
-            listOfThreads.add(thread {
+            listOfThreads.add(cachedThreadPool.submit(Thread {
```

Comment:
now using cached thread pools intead of raw threads, so we can receive thrown exceptions.  Before this, exceptions might happen in a thread and we wouldn't know unless we saw it happen in the log.